hibernate-core/src/main/java/org/hibernate/generator/Assigned.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or http://www.gnu.org/licenses/lgpl-2.1.html.
+ */
+package org.hibernate.generator;
+
+import org.hibernate.Internal;
+
+import java.util.EnumSet;
+
+/**
+ * A {@link Generator} that doesn't generate.
+ * <p>
+ * Identifier fields with an {@code Assigned} generator
+ * must be assigned a value by the application program.
+ * <p>
+ * This replaces the {@link org.hibernate.id.Assigned}
+ * identifier generation strategy from older versions
+ * of Hibernate.
+ *
+ * @since 7.0
+ *
+ * @author Gavin King
+ */
+@Internal
+public class Assigned implements Generator {
```

Comment:
Maybe this is a stupid question, but why do we need this class at all? Can't we just have a `null` generator when an id is assigned?

Reply:
> Maybe this is a stupid question

_Definitely_ not a stupid question; it's one I asked myself.

> Can't we just have a `null` generator when an id is assigned?

I think we could, yes, and I considered it.

However, I believe this works out simpler now that we have a few things which do almost-but-not-quite "assigned" generation: 

- the "foreign" generator used to implement `@MapsId`, and
- generators which do a combination of assignment and generation (this encompasses at least two different subcategories). 

And since in the new framework a `Generator` doesn't always generate: it can generate on insert, on update, on both, or on neither, this new object `Assigned` fits rather comfortably into the overall model.